data_structures/binary_trees/words_alphabetical.c

```diff
@@ -0,0 +1,255 @@
+ * @details
+ * More details about this
+ * [algorithm](http://www.dailyfreecode.com/Code/word-list-reads-text-file-makes-2050.aspx)
+ */
+#include <ctype.h>    /// for type checks
+#include <stdbool.h>  /// for boolean data type
+#include <stdio.h>    /// for IO operations
+#include <stdlib.h>   /// for memory allocation
+#include <string.h>   /// for string operations
+
+/**
+ * @brief structure defining a node in the binary tree
+ */
+struct Node
+{
+    char *word;              ///< the word (value) of the node
+    unsigned int frequency;  ///< number of occurences of the word
+    struct Node *left;       ///< pointer to the left child node
+    struct Node *right;      ///< pointer to the right child node
+};
+
+struct Node *readWordsInFileToTree(
+    FILE *file, struct Node *root);  ///< Reads words from file to tree and
+                                     ///< returns pointer to root node
+struct Node *addWordToTree(
+    char *word,
+    struct Node *currentNode);  ///< Adds word (node) to the correct position in
+                                ///< tree and returns pointer to root node
+void writeContentOfTreeToFile(
+    struct Node *node,
+    FILE *file);  ///< Writes contents of tree to another file alphabetically
+char *getPointerToWord(
+    char *word);  ///< Stores a word in memory and returns pointer to word
+struct Node *allocateMemoryForNode();  ///< Reserves memory for new node and
+                                       ///< returns pointer to node
+void closeFile(FILE *file);            ///< Closes file after read or write
+void freeTreeMemory(
+    struct Node *node);  ///< Frees memory when program is terminating
+void endProgramAbruptly(char *errorMessage);  ///< Ends program due to an error
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+    struct Node *root = NULL;  ///< pointer to root node
+    FILE *file = NULL;         ///< pointer to file
+
+    file = fopen("file.txt", "r");
+    if (file == NULL)
+        endProgramAbruptly("file: 'file.txt' not found");
+
+    root = readWordsInFileToTree(file, root);
+    closeFile(file);
+
+    file = fopen("wordcount.txt", "a");
+    fprintf(file, "%-5s \t %9s \t %s \n", "S/N", "FREQUENCY", "WORD");
+    writeContentOfTreeToFile(root, file);
+    closeFile(file);
+
+    freeTreeMemory(root);
+
+    return 0;
+}
+
+/**
+ * @brief Reads words from file to tree
+ * @param file file to be read from
+ * @param root root node of tree
+ * @returns a pointer to the root node
+ */
+struct Node *readWordsInFileToTree(FILE *file, struct Node *root)
+{
+    // longest english word = 45 chars
+    // +1 for '\0' = 46 chars
+    char *inputString =
+        (char *)malloc(46 * sizeof(char));  ///< pointer to input string
+
+    char inputChar;                ///< temp storage of characters
+    bool isPrevCharAlpha = false;  ///< bool to mark the end of a word
+    int pos = 0;  ///< position in inputString to place the inputChar
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. **Requires adding the `inttypes.h` library**. Check other parts of the code ([reference](https://en.cppreference.com/w/c/types/integer)). 🙂